From db54ce72d255b65307d7fafcdcf22df449e87db2 Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Mon, 21 Nov 2022 17:50:09 +0100
Subject: [PATCH 29/39] utils: add iwinfo_htmode_is_{ht|vht|he} helpers

Small and useful functions which allow to clear up some consumers.

Signed-off-by: Andre Heider <a.heider@gmail.com>
---
 include/iwinfo/utils.h |  3 +++
 iwinfo_utils.c         | 42 ++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 45 insertions(+)

diff --git a/include/iwinfo/utils.h b/include/iwinfo/utils.h
index 6fe5785..e9b8f1d 100644
--- a/include/iwinfo/utils.h
+++ b/include/iwinfo/utils.h
@@ -44,6 +44,9 @@ static inline int iwinfo_mbm2dbm(int gain)
 }
 
 size_t iwinfo_format_hwmodes(int modes, char *buf, size_t len);
+int iwinfo_htmode_is_ht(int htmode);
+int iwinfo_htmode_is_vht(int htmode);
+int iwinfo_htmode_is_he(int htmode);
 
 int iwinfo_ifup(const char *ifname);
 int iwinfo_ifdown(const char *ifname);
diff --git a/iwinfo_utils.c b/iwinfo_utils.c
index 63ae78c..c7713ed 100644
--- a/iwinfo_utils.c
+++ b/iwinfo_utils.c
@@ -102,6 +102,48 @@ size_t iwinfo_format_hwmodes(int modes, char *buf, size_t len)
 	return res;
 }
 
+int iwinfo_htmode_is_ht(int htmode)
+{
+	switch (htmode)
+	{
+	case IWINFO_HTMODE_HT20:
+	case IWINFO_HTMODE_HT40:
+		return 1;
+	}
+
+	return 0;
+}
+
+int iwinfo_htmode_is_vht(int htmode)
+{
+	switch (htmode)
+	{
+	case IWINFO_HTMODE_VHT20:
+	case IWINFO_HTMODE_VHT40:
+	case IWINFO_HTMODE_VHT80:
+	case IWINFO_HTMODE_VHT80_80:
+	case IWINFO_HTMODE_VHT160:
+		return 1;
+	}
+
+	return 0;
+}
+
+int iwinfo_htmode_is_he(int htmode)
+{
+	switch (htmode)
+	{
+	case IWINFO_HTMODE_HE20:
+	case IWINFO_HTMODE_HE40:
+	case IWINFO_HTMODE_HE80:
+	case IWINFO_HTMODE_HE80_80:
+	case IWINFO_HTMODE_HE160:
+		return 1;
+	}
+
+	return 0;
+}
+
 int iwinfo_ifup(const char *ifname)
 {
 	struct ifreq ifr;
-- 
2.35.1

